### PR TITLE
Issue 249, 247, 163, 207

### PR DESCRIPTION
**Pull request by RLiNREL**
This connects to,
#249 add conversion from non-standard energy unit to standard energy unit in facility unit settings, pre-assessment pages, industrial system pages
#247 fix the feedback update facility energy use on company unit change
#163 fix the feedback for industrial system page
#207 fix feedback for assessment pages

### CODE REVIEW

File: src/app/indexed-db/energy-opportunity-idb.service.ts [hunk 1]
```diff
@@ -59,4 +59,19 @@ export class EnergyOpportunityIdbService {
       return opportunity.guid == guid;
     });
   }
+
+  getByOtherGuid(guid: string, idType: string): Array<IdbEnergyOpportunity> {
```
Comment (rmroot) [inline src/app/indexed-db/energy-opportunity-idb.service.ts]: You can define the `idType: 'company' | 'facility'`

File: src/app/indexed-db/facility-idb.service.ts [hunk 2]
```diff
@@ -67,4 +67,18 @@ export class FacilityIdbService {
     await this.setFacilities();
     return newFacility.guid;
   }
+
+  getByOtherGuid(guid: string, type: string): Array<IdbFacility> {
```
Comment (rmroot) [inline src/app/indexed-db/facility-idb.service.ts]: `type: 'company' | 'user'`

File: src/app/setup-wizard/data-collection/on-site-assessment/assessment-details-form/assessment-details-form.component.ts [hunk 3]
```diff
@@ -57,7 +72,15 @@ export class AssessmentDetailsFormComponent {
...
-    })
+    });
+
+    this.companySub = this.companyIdbService.selectedCompany.subscribe(_company => {
+      this.companyEnergyUnit = _company.companyEnergyUnit;
+    });
+
+    this.facilitySub = this.facilityIdbService.selectedFacility.subscribe(_facility => {
+      this.facilityUnitSettings = _facility.unitSettings;
+    });
```
Comment (rmroot) [inline src/app/setup-wizard/data-collection/on-site-assessment/assessment-details-form/assessment-details-form.component.ts:83]: need to unsubscribe these in ngOnDestroy()

File: src/app/setup-wizard/data-collection/on-site-assessment/assessment-details-form/assessment-details-form.component.ts [hunk 4]
```diff
@@ -70,22 +93,46 @@ export class AssessmentDetailsFormComponent {
...
       _assessmentOption => _assessmentOption.assessmentType == this.assessment.assessmentType)?.utilityTypes || [];
     this.assessment.utilityTypes = utilityTypes; // track all utility types
-    // utility type is not the default if the assessment type changes
-    if (this.assessment.utilityType !== utilityTypes[0]) {
-      this.assessment.utilityType = utilityTypes?.[0]; // update to the first/default utility type
-      await this.utilityTypeChange();
-    } else {
-      await this.saveChanges();
-    }
+    this.calculateEnergyUseCost();
```
Comment (rmroot) [inline src/app/setup-wizard/data-collection/on-site-assessment/assessment-details-form/assessment-details-form.component.ts]: I think this should have an await

File: src/app/setup-wizard/data-collection/on-site-assessment/assessment-energy-opportunities-form/energy-opportunity-setup-form/energy-opportunity-setup-form.component.ts [hunk 5]
```diff
@@ -32,16 +33,24 @@ export class EnergyOpportunitySetupFormComponent {
...
     private setupWizardService: SetupWizardService,
-    private nonEnergyBenefitsIdbService: NonEnergyBenefitsIdbService
+    private nonEnergyBenefitsIdbService: NonEnergyBenefitsIdbService,
+    private companyIdbService: CompanyIdbService,
   ) {
   }
 
   ngOnInit() {
     this.energyOpportunity = this.energyOpportunityIdbService.getByGuid(this.energyOpportunityGuid);
+    this.companySub = this.companyIdbService.selectedCompany.subscribe(company => {
```
Comment (rmroot) [inline src/app/setup-wizard/data-collection/on-site-assessment/assessment-energy-opportunities-form/energy-opportunity-setup-form/energy-opportunity-setup-form.component.ts:51]: need to unsubscribe this in ngOnDestroy

File: src/app/setup-wizard/pre-visit/facility-energy-equipment-setup/energy-equipment-form/energy-equipment-form.component.ts [hunk 6]
```diff
@@ -43,16 +46,25 @@ export class EnergyEquipmentFormComponent {
...
+    private companyIdbService: CompanyIdbService,
   ) { }
 
   ngOnInit() {
     this.energyEquipment = this.energyEquipmentIdbService.getByGuid(this.energyEquipmentGuid);
     this.contactSub = this.contactIdbService.contacts.subscribe(_contacts => {
       this.contacts = _contacts;
     });
+
+    this.companySub = this.companyIdbService.selectedCompany.subscribe(_company => {
```
Comment (rmroot) [inline src/app/setup-wizard/pre-visit/facility-energy-equipment-setup/energy-equipment-form/energy-equipment-form.component.ts:65]: need to unsubscribe subscription on destroy

File: src/app/setup-wizard/pre-visit/pre-assessment-setup/pre-assessment-setup.component.html [hunk 7]
```diff
@@ -126,6 +126,36 @@ <h6>Utility Types</h6>
                                                                     </div>
                                                                 </div>
                                                             </ng-container>
+                                                            <ng-container *ngIf="use.include
+                                                                && !(use.utilityType | isStandardUnit :use.energyUnit : 'Energy')">
+                                                                <div class="col-4">
+                                                                    <div class="input-group">
```
Comment (rmroot) [inline src/app/setup-wizard/pre-visit/pre-assessment-setup/pre-assessment-setup.component.html:132]: Make sure this label is lined up with the label above it. Right now it's off by a bit

File: src/app/setup-wizard/pre-visit/pre-assessment-setup/pre-assessment-setup.service.ts [hunk 8]
```diff
@@ -45,7 +73,7 @@ export class PreAssessmentSetupService {
           let camelCaseType = trimmedType.charAt(0).toLowerCase() + trimmedType.slice(1);
           let convertedCost = this.convertValue.convertValue(
```
Comment (rmroot) [inline src/app/setup-wizard/pre-visit/pre-assessment-setup/pre-assessment-setup.service.ts:74]: Converting costs aren't working properly. as a rule of thumb, if you change the energy unit the resulting cost should not change if the conversions are done correctly

File: src/app/shared/constants/unitOptions.ts [hunk 9]
```diff
@@ -177,80 +178,95 @@ export const VolumeGasOptions: Array<UnitOption> = [
     },
 ]
 
-export const SteamUnitOptions: Array<UnitOption> = [
+export const EnergySteamUnitOptions: Array<UnitOption> = [
```
Comment (rmroot) [inline src/app/shared/constants/unitOptions.ts:181]: "lb" isn't an energy unit

File: src/app/shared/shared-settings-forms/units-form/units-form.component.html [hunk 10]
```diff
@@ -300,6 +369,29 @@
                 </div>
             </div>
         </ng-container>
+        <ng-container *ngIf="form.controls['includeCompressedAir'].value
+            && !('Compressed Air' |isStandardUnit :form.controls['compressedAirUnit'].value : 'Energy')">
+            <div class="col-4">
+                <div class="input-group">
+                    <input type="text" class="disabled-input-label-secondary" [disabled]="true"
```
Comment (rmroot) [inline src/app/shared/shared-settings-forms/units-form/units-form.component.html:376]: Please make sure these labels line up properly

**Review by rmroot — CHANGES_REQUESTED** [review]

File: src/app/shared/constants/unitOptions.ts (hunk 9 above)
Comment (RLiNREL) [reply src/app/shared/constants/unitOptions.ts:181]: yes, it is kristina's request to have mixed non-standard energy unit with standard energy unit for steam (https://github.com/NREL/NEB-Tool/issues/207). I lumped them together as steam unit options and treat them differently when pull the units in. Perhaps there is a better way to store them separately. I will create a new issue to think about it later.

Comment (RLiNREL) [reply src/app/shared/constants/unitOptions.ts:181]: issue 272 was created, marked resolved, please let me know if this works

**Review by rmroot — APPROVED** [review]

**Pull request merged**